src/lib/controllers/ui-controller.js

```diff
@@ -1051,4 +1103,117 @@ export default class UIController {
+            langOpts: { [Constants.LANG_PERSIAN]: { lookupMorphLast: true } } // TODO this should be externalized
+          }),
+          {
+            experience: 'Get word data',
+            actions: [
+              { name: 'getData', action: ExpObjMon.actions.START, event: ExpObjMon.events.GET },
+              { name: 'finalize', action: ExpObjMon.actions.STOP, event: ExpObjMon.events.GET }
+            ]
+          })
+          .getData() */
+
+        LexicalQuery.create(textSelector, {
+          htmlSelector: htmlSelector,
+          uiController: this,
+          maAdapter: this.maAdapter,
+          lexicons: Lexicons,
+          resourceOptions: this.resourceOptions,
+          siteOptions: [],
+          lemmaTranslations: this.enableLemmaTranslations(textSelector) ? { adapter: LemmaTranslations, locale: this.options.items.locale.currentValue } : null,
+          langOpts: { [Constants.LANG_PERSIAN]: { lookupMorphLast: true } } // TODO this should be externalized
+        }).getData()
+      }
+    }
+  }
+
+  /**
+   * Check to see if Lemma Translations should be enabled for a query
+   *  NB this is Prototype functionality
+   */
+  enableLemmaTranslations (textSelector) {
+    return textSelector.languageID === Constants.LANG_LATIN &&
+      this.options.items.enableLemmaTranslations.currentValue &&
+      !this.options.items.locale.currentValue.match(/^en-/)
+  }
+
+  handleEscapeKey (event) {
+    // TODO: Move to keypress as keyCode is deprecated
+    // TODO: Why does it not work on initial panel opening?
+    if (event.keyCode === 27 && this.state.isActive()) {
+      if (this.state.isPanelOpen()) {
+        this.panel.close()
+      } else if (this.popup.visible) {
+        this.popup.close()
+      }
+    }
+    return true
+  }
+
+  handleReload () {
+    console.log('Alpheios reload event caught.')
+    if (this.state.isActive()) {
+      this.deactivate()
+    }
+    window.location.reload()
+  }
+
+  disableContent () {
+    console.log('Alpheios is embedded.')
+    // if we weren't already disabled, remember the current state
+    // and then deactivate before disabling
+    if (!this.state.isDisabled()) {
+      this.state.save()
+      if (this.state.isActive()) {
+        console.log('Deactivating Alpheios')
+        this.deactivate()
+      }
+    }
+    this.state.disable()
+    // TODO: Need to handle this in a content. Send state to BG on every state change?
+    // this.sendStateToBackground()
+  }
+
+  /**
+   * Issues an AnnotationQuery to find and apply annotations for the currently loaded document
+   */
+  updateAnnotations () {
+    if (this.state.isActive() && this.state.uiIsActive()) {
+      AnnotationQuery.create({
+        uiController: this,
+        document: document,
+        siteOptions: this.siteOptions
+      }).getData()
+    }
+  }
+
+  updatePanelOnActivation () {
```

Comment:
see prior note. This behavior is probably application specific.

Reply:
OK